### PR TITLE
WLP improvements [6168]

### PR DESCRIPTION
**Pull request by raquelalvarezbanos**
This PR:

* Fixes a bug in the return value when using security in WLP
* Optimizes removal of local writers
* Tries to make liveliness tests more stable

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/198/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/181/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/197/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/204/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/186/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/203/)

File: src/cpp/rtps/builtin/discovery/participant/PDPClient.cpp [hunk 1]
```diff
@@ -168,6 +168,8 @@ bool PDPClient::initPDP(RTPSParticipantImpl* part)
         return false;
     }
 
+    enable();
```
Comment (MiguelCompany) [inline src/cpp/rtps/builtin/discovery/participant/PDPClient.cpp]: Shouldn't this be done after initialization of WLP?

File: src/cpp/rtps/builtin/liveliness/WLPListener.cpp [hunk 2]
```diff
@@ -65,21 +65,30 @@ void WLPListener::onNewCacheChangeAdded(
...
         }
     }
-    if(change->serializedPayload.length>0)
+    if (change->serializedPayload.length > 0)
     {
-        for(uint8_t i =0;i<12;++i)
+        if (PL_CDR_BE == change->serializedPayload.data[1])
         {
-            guidP.value[i] = change->serializedPayload.data[i];
+        change->serializedPayload.encapsulation = (uint16_t)PL_CDR_BE;
```
Comment (MiguelCompany) [inline src/cpp/rtps/builtin/liveliness/WLPListener.cpp]: indentation

File: src/cpp/rtps/builtin/liveliness/WLPListener.cpp [hunk 3]
```diff
@@ -131,7 +140,7 @@ bool WLPListener::computeKey(CacheChange_t* change)
         SerializedPayload_t* pl = &change->serializedPayload;
         if(pl->length >= 16)
```
Comment (MiguelCompany) [inline src/cpp/rtps/builtin/liveliness/WLPListener.cpp]:
```suggestion
        if(pl->length >= 20)
```

File: src/cpp/rtps/builtin/BuiltinProtocols.cpp [hunk 4]
```diff
@@ -125,6 +125,11 @@ bool BuiltinProtocols::initBuiltinProtocols(
         mp_WLP->initWL(mp_participantImpl);
     }
 
+    if(m_att.discovery_config.discoveryProtocol == DiscoveryProtocol_t::SIMPLE)
```
Comment (MiguelCompany) [inline src/cpp/rtps/builtin/BuiltinProtocols.cpp]: I think this should also be done for CLIENT. See my other comment

File: src/cpp/rtps/builtin/liveliness/WLP.cpp [hunk 5]
```diff
@@ -157,7 +157,10 @@ bool WLP::initWL(RTPSParticipantImpl* p)
 
     bool retVal = createEndpoints();
 #if HAVE_SECURITY
-    if (retVal) createSecureEndpoints();
+    if (retVal)
+    {
+        retVal = retVal && createSecureEndpoints();
```
[the remark this answers is not in the document]
Comment (raquelalvarezbanos) [reply src/cpp/rtps/builtin/liveliness/WLP.cpp]: :+1:

File: src/cpp/rtps/builtin/discovery/participant/PDPClient.cpp (hunk 1 above)
Comment (raquelalvarezbanos) [reply src/cpp/rtps/builtin/discovery/participant/PDPClient.cpp]: I agree, changed.

File: src/cpp/rtps/builtin/liveliness/WLPListener.cpp (hunk 2 above)
Comment (raquelalvarezbanos) [reply src/cpp/rtps/builtin/liveliness/WLPListener.cpp]: :+1:

File: src/cpp/rtps/builtin/liveliness/WLPListener.cpp (hunk 3 above)
Comment (raquelalvarezbanos) [reply src/cpp/rtps/builtin/liveliness/WLPListener.cpp]: :+1:

File: src/cpp/rtps/builtin/BuiltinProtocols.cpp (hunk 4 above)
Comment (raquelalvarezbanos) [reply src/cpp/rtps/builtin/BuiltinProtocols.cpp]: Updated

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/222/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/201/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/221/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/226/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/205/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/225/)

**Comment by richiware** [comment]
Failed tests not related.

**Pull request merged**